app/Main.hs

```diff
@@ -114,6 +134,25 @@ makeCard c = do
                 xs <- getLines
                 return (x:xs)
 
+        getCardType :: Maybe Card -> IO Char
+        getCardType d = do
+            putStr "\nEnter Card Type:"
+            let defaultVal = case d of
+                        Just (CardProgram _ _ _ _ _ _ _) -> Just 'P'
```

Comment:
It might be nice to use record wildcards for this:
```suggestion
                        Just CardProgram{..} -> Just 'P'
```
That will require `RecordWildCards` to be added to `package.yaml` under `default-extensions`.

Reply:
I knew there must be a neater way I was missing, done with https://github.com/fpclass/perpetual-haskelling-initiative/pull/15/commits/2db0f547154d4c73b30d91a731cc009b08022988